flake.nix

```diff
@@ -153,6 +149,27 @@
         buildInputs = [ ] ++ extraBuildInputs;
       };
 
+      # The `port` is parameterized to support parallel test runs without colliding static servers
+      networkTestArgs = port: {
+        # We provide `barretenberg-transcript00` from the overlay to the tests as a URL hosted via a static server
+        # This is necessary because the Nix sandbox has no network access and downloading during tests would fail
+        TRANSCRIPT_URL = "http://0.0.0.0:${toString port}/${builtins.baseNameOf pkgs.barretenberg-transcript00}";
+
+        # This copies the `barretenberg-transcript00` from the Nix store into this sandbox
+        # which avoids exposing the entire Nix store to the static server it starts
+        # The static server is moved to the background and killed after checks are completed
+        preCheck = ''
+          cp ${pkgs.barretenberg-transcript00} .
+          echo "Starting simple static server"
+          ${pkgs.simple-http-server}/bin/simple-http-server --port ${toString port} --silent &
+          HTTP_SERVER_PID=$!
+        '';
+
+        postCheck = ''
+          kill $HTTP_SERVER_PID
+        '';
+      };
```

Comment:
The network/dns failures on linux were due to the Nix sandbox disabling networking (it doesn't disable networking on mac for historical reasons).

So I needed to mimic our downloading locally in the sandbox, as seen in this code. Copying @kobyhallx for review, since it was his guidance in determining a solution.

Reply:
This looks good.